test/python/common/fio_spdk.py

```diff
@@ -0,0 +1,32 @@
+import asyncio
+import os
+import shutil
+from common.command import run_cmd_async
+from urllib.parse import urlparse
+
+
+class FioSpdk(object):
+
+    def __init__(self, name, rw, uri, runtime=15):
+        self.name = name
+        self.rw = rw
+        u = urlparse(uri)
+        self.host = u.hostname
+        self.port = u.port
+        self.nqn = u.path[1:]
+        self.cmd = shutil.which("fio")
+        self.runtime = runtime
+
+    async def run(self):
+        spdk_path = os.environ.get('SPDK_PATH')
+        if spdk_path is None:
+            spdk_path = os.getcwd() + '/../../spdk-sys/spdk/build'
+        command = ("sudo LD_PRELOAD={}/fio/spdk_nvme fio --ioengine=spdk "
```

Comment:
if this is using the spdk plugin that issues IO directly (bypassing the kernel), does it still need sudo?

Reply:
Yes, because SPDK needs it. This part isn't used in this PR anyway, I'll raise another PR later which also refactors this.